### PR TITLE
Xcode Archive action now works.

### PR DESCRIPTION
**Pull request by ufoxp**
1.  Framework is now linked in more elastic way.
2.  Application INSTALL_PATH is now setup properly.
3.  Framework copy / sign post build commands are now more elastic.
4.  SKIP_INSTALL is now set to YES on framework target as it should be.

### CODE REVIEW

**Comment by ufoxp** [comment]
This should take care of #6.

File: cppframework/CMakeLists.txt [hunk 1]
```diff
@@ -27,18 +27,19 @@ set_target_properties(${FRAMEWORK_NAME} PROPERTIES
...
-    COMMAND /bin/bash -c "${CMAKE_CURRENT_LIST_DIR}/install_name.sh $<TARGET_FILE:${FRAMEWORK_NAME}>"
+    COMMAND /bin/bash -c "${CMAKE_CURRENT_LIST_DIR}/install_name.sh \${BUILT_PRODUCTS_DIR}/\${PRODUCT_NAME}.framework/\${PRODUCT_NAME}"
 )
 
 add_custom_command(
     TARGET ${FRAMEWORK_NAME}
     POST_BUILD
-    COMMAND install_name_tool -id \"@rpath/${FRAMEWORK_NAME}.framework/${FRAMEWORK_NAME}\"
-    $<TARGET_FILE:${FRAMEWORK_NAME}>
+    COMMAND install_name_tool -id \"@rpath/\${PRODUCT_NAME}.framework/\${PRODUCT_NAME}\"
```
Comment (sheldonth) [inline cppframework/CMakeLists.txt]: Why are you using the PRODUCT_NAME here? The FRAMEWORK_NAME is a seperate value from the PRODUCT_NAME so you can have an app called foobar and  a framework called bar inside it.

**Comment by sheldonth** [comment]
>  framework target as it should be.

What is accomplished with this?

File: CMakeLists.txt [hunk 2]
```diff
@@ -118,45 +118,97 @@ set_target_properties(${APP_NAME} PROPERTIES
...
                       XCODE_ATTRIBUTE_TARGETED_DEVICE_FAMILY "1"
                       XCODE_ATTRIBUTE_CLANG_ENABLE_OBJC_ARC YES
-                      XCODE_ATTRIBUTE_COMBINE_HIDPI_IMAGES "NO")
-                      
+                      XCODE_ATTRIBUTE_COMBINE_HIDPI_IMAGES "NO"
+                      XCODE_ATTRIBUTE_INSTALL_PATH "$(LOCAL_APPS_DIR)")
 
-#include framework headers
+# Include framework headers, needed to make "Build" Xcode action work.
+# "Archive" works fine just relaying on default search paths as it
```
Comment (sheldonth) [inline CMakeLists.txt]: English semantic nit: relaying != relying

**Comment by sheldonth** [comment]
More reading and I think I've come to see that `PRODUCT_NAME` is a runtime variable actually defined by XCode. And the subsitution occurs because the cmake custom_command is invoked through Xcode? Is that your understanding of that? (re: line 42 cppframework/cmakelists.txt)

Note that if you add a line like `MESSAGE(STATUS ${PRODUCT_NAME})` right before the install_name_tool shell invocation it does print the app name, e.g. YOURAPP yet it's has the correct interpolation in the shell invocation. Anyways, I am super excited by this upgrade. Excellent work, thanks a bunch for this.

File: cppframework/CMakeLists.txt (hunk 1 above)
Comment (ufoxp) [inline cppframework/CMakeLists.txt]: Target to which I we add this command is framework not app. PRODUCT_NAME changes between targets... and this command runs in Xcode not CMake.

File: CMakeLists.txt (hunk 2 above)
Comment (ufoxp) [inline CMakeLists.txt]: Fixed.

**Comment by ufoxp** [comment]
Without SKIP_INSTALL application archive will have framework under products - even with full path... so there will be path Archive/Products/Users/something/something/etc/Xcode/cppframework/Release-iphoneos/FooBar.framework.

**Comment by sheldonth** [comment]
Excellent. Closes #6

**Pull request merged**